pkg/sql/gcjob/gcjobnotifier/notifier.go

```diff
@@ -185,10 +195,16 @@ func (n *Notifier) maybeNotify() {
 	if !zoneConfigUpdated {
 		return
 	}
-	n.notify()
+	n.notifyLocked()
 }
 
 func (n *Notifier) notify() {
```

Comment:
Now all call sites of `notify` must not have already acquired the mutex. Tests should be able to verify that, right?

Reply:
I added this because of a race detector failure. In practice that call to notify based on the upgrade seemed to not really happen in a racy way. There are no other callers of notify. What's the ask here?

Reply:
there is no ask here :) it's more of a clarification question. Thanks!